invokeai/backend/quantization/gguf/loaders.py

```diff
@@ -0,0 +1,68 @@
+# Largely based on https://github.com/city96/ComfyUI-GGUF
+
+from pathlib import Path
+
+import gguf
+import torch
+
+from invokeai.backend.quantization.gguf.layers import GGUFTensor
+from invokeai.backend.quantization.gguf.utils import detect_arch
+
+
+def gguf_sd_loader(
+    path: Path, handle_prefix: str = "model.diffusion_model.", data_type: torch.dtype = torch.bfloat16
+) -> dict[str, GGUFTensor]:
+    """
+    Read state dict as fake tensors
+    """
+    reader = gguf.GGUFReader(path)
+
+    prefix_len = len(handle_prefix)
+    tensor_names = {tensor.name for tensor in reader.tensors}
+    has_prefix = any(s.startswith(handle_prefix) for s in tensor_names)
+
+    tensors: list[tuple[str, gguf.ReaderTensor]] = []
+    for tensor in reader.tensors:
+        sd_key = tensor_name = tensor.name
+        if has_prefix:
+            if not tensor_name.startswith(handle_prefix):
+                continue
+            sd_key = tensor_name[prefix_len:]
+        tensors.append((sd_key, tensor))
+
+    # detect and verify architecture
+    compat = None
+    arch_str = None
+    arch_field = reader.get_field("general.architecture")
+    if arch_field is not None:
+        if len(arch_field.types) != 1 or arch_field.types[0] != gguf.GGUFValueType.STRING:
+            raise TypeError(f"Bad type for GGUF general.architecture key: expected string, got {arch_field.types!r}")
+        arch_str = str(arch_field.parts[arch_field.data[-1]], encoding="utf-8")
+        if arch_str not in {"flux"}:
+            raise ValueError(f"Unexpected architecture type in GGUF file, expected flux, but got {arch_str!r}")
+    else:
+        arch_str = detect_arch({val[0] for val in tensors})
```

Comment:
The first argument in the call signature for `detect_arch()` was originally `dict[str, Tensor]`, but here you're passing a set of str. Since the function is only called here, I proposed a change in the `detect_arch` function below.